packages/apps/docs/src/components/Markdown/Heading/Heading.tsx

```diff
@@ -93,12 +44,54 @@ export const TaggedHeading: FC<IProp> = ({
 
   const slug = createSlug(slugInputStr, index, parentTitle);
 
-  return (
-    <StyledHeader as={as} variant={variant ?? as}>
-      {children}
-      <StyledLinkIcon id={slug} href={`#${slug}`}>
-        <SystemIcons.Link />
-      </StyledLinkIcon>
-    </StyledHeader>
+  const ArticleLink = (
+    <a className={headerIconLinkClass} id={slug} href={`#${slug}`}>
+      <SystemIcon.Link />
+    </a>
   );
+  switch (as) {
+    case 'h2':
+      return (
+        <h2 className={headerClassVariants[variant ?? as]}>
```

Comment:
another improvement to make this better is move `headerClassVariants[variant ?? as]` to a variable so you can change it in one place rather than copy it to each of the classNames